Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -148,6 +148,16 @@ - (void)beginLocationUpdates
   [_locationManager startUpdatingLocation];
 }
 
+- (BOOL) isNewLocation:(CLLocation*) location
+{
+  if ([_lastLocationEvent objectForKey:@"coords"] != [NSNull null]){
+    if([_lastLocationEvent[@"coords"][@"latitude"] doubleValue] != location.coordinate.latitude || [_lastLocationEvent[@"coords"][@"longitude"] doubleValue] != location.coordinate.longitude){
```

Comment:
Split this over two lines and return the expression immediately.

In general, comparing floating point numbers directly is not a good idea, you should always compare the difference to a small episilon: http://stackoverflow.com/questions/10334688/how-dangerous-is-it-to-compare-floating-point-values